types/transaction.go

```diff
@@ -115,6 +115,10 @@ type Transaction interface {
 
 	// ResponseBodyWriter returns a io.Writer for writing the response body to.
 	// Contents will be buffered until the transaction is closed.
+	// Note that the Write function returns the number of written bytes. This
```

Comment:
I would drop this note here. We will most likely add a similar function for Response body.